finagle-core/src/main/scala/com/twitter/finagle/Stack.scala

```diff
@@ -455,6 +455,11 @@ object Stack {
       Node(this, (prms, next) => Leaf(this, make(next.make(prms))), next)
   }
 
+  class NoOpModule[T](val role: Role, val description: String) extends Module0[T] {
```

Comment:
I think you can avoid defining a new type by just passing an `identity` function as a stack module - it should be implicitly converted. We do that in [a bunch of places](https://github.com/twitter/finagle/blob/develop/finagle-core/src/main/scala/com/twitter/finagle/client/StackClient.scala#L70).